src/Paket.Core/Common/AssemblyMetadata.fs

```diff
@@ -0,0 +1,74 @@
+module Paket.AssemblyMetadata
+
+open Chessie.ErrorHandling
+open System
+open System.IO
+open System.Globalization
+open System.Reflection
+open System.Reflection
+open System.Reflection.Metadata
+open System.Reflection.PortableExecutable
+
+/// Calculates the short form of the public key token for use with binding redirects, if it exists.
+let getPublicKeyToken (assembly:AssemblyName) =
+    ("", assembly.GetPublicKeyToken())
+    ||> Array.fold(fun state b -> state + b.ToString("X2"))
+    |> function
+    | "" -> None
+    | token -> Some (token.ToLower())
+
+/// Creates AssemblyName of AssemblyReference opened on the MetadataReader; 
+/// expected to be part of next release of the Metadata support, so we can remove this. 
+/// Can throw AccessViolation if used incorrectly, with unrelated or disposed reader. 
+let private getAssemblyName (reader:MetadataReader) (reference:AssemblyReference) = 
+    let assemblyName = reader.GetString(reference.Name)
+    let cultureInfo = 
+        if reference.Culture.IsNil then null 
+        else CultureInfo(reader.GetString(reference.Culture))
+    let hasPublicKey = reference.Flags.HasFlag(AssemblyFlags.PublicKey)
+    let assemblyFlag = 
+        if hasPublicKey then AssemblyNameFlags.PublicKey else AssemblyNameFlags.None
+    let assemblyName = 
+        new AssemblyName(
+            Name = assemblyName,
+            Flags = assemblyFlag,
+            Version = reference.Version,
+            CultureInfo = cultureInfo)
+    let keyOrToken = 
+        let keyOrTokenHandle = reference.PublicKeyOrToken
+        if keyOrTokenHandle.IsNil then null else reader.GetBlobBytes(keyOrTokenHandle)
+    if hasPublicKey then
+        assemblyName.SetPublicKey(keyOrToken)
+    else 
+        assemblyName.SetPublicKeyToken(keyOrToken)
+    assemblyName;
```

Comment:
semicolon?

Reply:
still new to F# :) is this incorrect? I prefer using semicolon when using a variable like this, just to make it the return value

Reply:
usually we leave out semicolons ;)